packages/components/src/components/Button/Button.tsx

```diff
@@ -90,8 +90,13 @@ export const Button = forwardRef<HTMLButtonElement | null, ButtonProps>(
                     iconStyles({ style, ...props }),
                     className,
                 )}
-                onClick={onPress}
                 {...props}
+                onClick={(e) => {
+                    // @ts-expect-error - this is done to allow other components who use Button as a trigger to pass an action
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+                    props.onClick?.(e);
+                    onPress(e);
```

Comment:
`onPress` could be undefined no?
```suggestion
                    onPress?.(e);
```